google/cloud/spanner/samples/postgresql_samples.cc

```diff
@@ -207,6 +215,444 @@ void DmlGettingStartedUpdate(google::cloud::spanner::Client client) {
 }
 // [END spanner_postgresql_dml_getting_started_update]
 
+// [START spanner_postgresql_batch_dml]
+void BatchDml(google::cloud::spanner::Client client) {
+  auto commit = client.Commit(
+      [&client](google::cloud::spanner::Transaction const& txn)
+          -> google::cloud::StatusOr<google::cloud::spanner::Mutations> {
+        std::vector<google::cloud::spanner::SqlStatement> sql = {
+            google::cloud::spanner::SqlStatement(
+                "INSERT INTO Singers (SingerId, FirstName, LastName)"
+                "    VALUES ($1, $2, $3)",
+                {{"p1", google::cloud::spanner::Value(3)},
+                 {"p2", google::cloud::spanner::Value("Alice")},
+                 {"p3", google::cloud::spanner::Value("Trentor")}}),
+            google::cloud::spanner::SqlStatement(
+                "INSERT INTO Singers (SingerId, FirstName, LastName)"
+                "    VALUES ($1, $2, $3)",
+                {{"p1", google::cloud::spanner::Value(4)},
+                 {"p2", google::cloud::spanner::Value("Lea")},
+                 {"p3", google::cloud::spanner::Value("Martin")}}),
+        };
+        auto result = client.ExecuteBatchDml(txn, std::move(sql));
+        if (!result) return std::move(result).status();
+        for (std::size_t i = 0; i < result->stats.size(); ++i) {
+          std::cout << result->stats[i].row_count << " row(s) affected"
+                    << " for statement " << (i + 1) << ".\n";
+        }
+        // Batch operations may have partial failures, in which case
+        // ExecuteBatchDml() returns with success, but the application
+        // should verify that all statements completed successfully.
+        if (!result->status.ok()) return result->status;
+        return google::cloud::spanner::Mutations{};
+      });
+  if (!commit) throw std::runtime_error(commit.status().message());
+  std::cout << "Update was successful.\n";
+}
+// [END spanner_postgresql_batch_dml]
+
+void DropTables(google::cloud::spanner_admin::DatabaseAdminClient client,
```

Comment:
Seems odd that there are no region tags or doxygen tags for this sample.  Do we really need it?

Reply:
We only need it to clear state for the subsequent samples, which want to recreate these tables in a different way.  Indeed, the "auto run" calls `DropTables()` twice.  There is nothing PostgreSQL specific about it though, so there is no "spanner_postgresql_*" tag defined for dropping tables.

I did make up such a tag for the "banner" line output however, but I could just as easily get rid of that, or replace it with something less tag-like.

(Some of the other languages completely create a new database, and create and populate tables, before every sample and according to its needs, and then tear the whole thing down again before starting the next sample, but that is really slow.)

Reply:
Ack.

Reply:
PS: `CreateTables()` falls into the same category.